### PR TITLE
Us378 rebase

### PR DESCRIPTION
**Pull request by dzadroga**
Completes US378
Tasks:
- Create card popup UI canvas/panel prefab (Task 379)
- Add UI elements (Card, background, text display, button) (Task 380)
- Create or implement event channel for "CardDrawn" event (Task 381)
- Wire CardUI to subscribe to CardDrawn Events (Task 382)
- Implement ShowCard() method to display card data (Task 383)
- Wire dismiss button to execute card and hide UI (Task 384)
- Add basic show/hide animations (Fadein/out or scale) (Task 385)
- Test with example cards  (Task 386)
- Verify UI blocks player input while displayed (Task 387)

This is definitely going to need to be integrated better within the next sprint, the basic functionality is all complete but the UI scene itself may need to be changed based on how much information is displayed on the cards.  Also, as the effects are added, the card may need to be updated.  Right now I left space for an image on the card similar to real monopoly but we can delete the if we want to.  There is a base UI test file that I included but has no logic right now, I wanted to leave it in case we decide to add more tests during integrations next sprint.

### CODE REVIEW

**Comment by kescob0825** [comment]
# Code Review Checklist

**Completed for**: US #378 and Tasks #379 to #387
**Code Review Completed By**: Karl Escobar

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [ ] Code is testable.
- [ ] Automated tests have been added or updated.
- [ ] Existing tests sufficiently cover the change (unit/integration/system).
- [ ] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---
Notes: Failing unit tests:
- InstrumentSpaceDataTests
- LaunchPadSpaceDataTests
- OwnableSpaceDataTests
- PlanetSpaceDataTests
- PropertySpaceDataTests

**Review by tsaldrid — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #378 and Tasks #379 to387
**Code Review Completed By**: @tsaldrid

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

I did not have the failing tests on my end for this that were in the other comment.

File: Assets/Scripts/Events/EventChannelTypes/CardDrawnEventChannel.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+using System;
+using UnityEngine;
+
+[CreateAssetMenu(
+    fileName = "CardDrawnEventChannel",
+    menuName = "Events/Card Drawn Event Channel")]
+public class CardDrawnEventChannel : ScriptableObject
+{
+    public event Action<Card, Player, CardDeck> OnCardDrawn;
```
Comment (hdathert) [inline Assets/Scripts/Events/EventChannelTypes/CardDrawnEventChannel.cs:9]: This definitely works, but is not aligned with our established pattern. We can refactor this later on though.

The pattern is to have our event payloads as their own class in `Events/EventDataStructures/`, not directly embedded into the channel itself.

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #378 and Tasks #379 to #387  
**Code Review Completed By**: @hdathert 
Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

> All of the tests were passing on my end as well.
